src/agent/debugger/src/target.rs

```diff
@@ -272,24 +207,159 @@ impl Target {
         self.modules.iter()
     }
 
-    pub fn initial_bp(&mut self, load_symbols: bool) -> Result<()> {
+    pub fn initial_bp(&mut self) -> Result<()> {
         self.saw_initial_bp = true;
 
-        if load_symbols || !self.breakpoints.is_empty() {
-            self.sym_initialize()?;
+        if !self.unresolved_breakpoints.is_empty() {
+            self.maybe_sym_initialize()?;
+            self.try_resolve_all_unresolved_breakpoints();
+            for module in self.modules.values_mut() {
+                module.apply_breakpoints(self.process_handle)?;
+            }
+        }
 
-            for (_, module) in self.modules.iter_mut() {
-                if let Err(e) = module.sym_load_module(self.process_handle) {
-                    error!("Error loading symbols: {}", e);
+        Ok(())
+    }
+
+    pub fn initial_wow64_bp(&mut self) {
+        self.saw_initial_wow64_bp = true;
+    }
+
+    fn try_resolve_all_unresolved_breakpoints(&mut self) {
+        // borrowck - take ownership from self so we call `try_resolve_unresolved_breakpoint`.
+        let mut unresolved_breakpoints = std::mem::take(&mut self.unresolved_breakpoints);
```

Comment:
Noting that we're allocating a new (empty) vec each time we do this, and could avoid that if we used `Option::take()`. But we only do this once per breakpoint, so it is no doubt lost in the perf noise.

Reply:
An empty Vec does not allocate, and using an option requires additional checks when pushing, so I'm going to leave this as is.

Reply:
Ah, right! This is def better, then.